src/Modules/CalcPerform.lua

```diff
@@ -3093,6 +3093,95 @@ function calcs.perform(env, avoidCache)
 		end
 	end
 
+	-- Doom Blast (from Impending Doom)
+	if env.player.mainSkill.skillData.triggeredWhenHexEnds and not env.player.mainSkill.skillFlags.minion then
+		local spellCount = {}
+		local source = nil
+		local hexCastRate = 0
+		local trigRate = 0
+		for _, skill in ipairs(env.player.activeSkillList) do
+			local match1 = env.player.mainSkill.activeEffect.grantedEffect.fromItem and skill.socketGroup.slot == env.player.mainSkill.socketGroup.slot
+			local match2 = (not env.player.mainSkill.activeEffect.grantedEffect.fromItem) and skill.socketGroup == env.player.mainSkill.socketGroup
+			if skill.skillTypes[SkillType.Hex] and (match1 or match2) then
+				source, hexCastRate = findTriggerSkill(env, skill, source, trigRate)
```

Comment:
Not sure what you were going for here but if you want to get the hex with the fastest cast rate you need to replace `trigRate` (just remove it) with `hexCastRate` as `findTriggerSkill` uses that var as the "highest rate so far".

Reply:
To be honest, I was going for copy paste, hehe. I'll clean it up!